src/main/navigation/navigation.c

```diff
@@ -1554,9 +1554,26 @@ void updateActualAltitudeAndClimbRate(bool estimateValid, float newAltitude, flo
  *-----------------------------------------------------------*/
 void updateActualHeading(bool headingValid, int32_t newHeading)
 {
-    /* Update heading */
+    /* Update heading. Check if we're acquiring a valid heading for the
+     * first time and update home heading accordingly.
+     */
+    navigationEstimateStatus_e newEstHeading = headingValid ? EST_TRUSTED : EST_NONE;
+    if (newEstHeading >= EST_USABLE && posControl.flags.estHeadingStatus < EST_USABLE &&
+        (posControl.homeFlags & (NAV_HOME_VALID_XY | NAV_HOME_VALID_Z)) &&
+        (posControl.homeFlags & NAV_HOME_VALID_HEADING) == 0) {
+
+        int32_t yawRotation = newHeading - posControl.actualState.yaw;
+        posControl.homePosition.yaw += yawRotation;
```

Comment:
Shouldn't we simply rewrite `posControl.homePosition.yaw` here? Why the `yawRotation` magic?

Reply:
You need the rotation because you might arm, turn and then launch. If you turned 90º before launching the "fake" yaw at that point will be 90º and home will be at 0º (assuming you armed without rotating since battery was plugged in). Now you take off and once you get the real heading yaw is becoming 180º. That means your fake yaw was 90º off to the left, hence your home yaw needs that same correction since it was calculated with the fake one.

The problem with storing the first acquired yaw as home yaw is that you might unintentionally throw the plane a bit off from the direction you intended or even the pitch/roll stabilisation might send it a bit to one side (launch mode doesn't try to keep heading, but level). Using the yaw when you arm allows more predictable values for the pilot.

Reply:
Ok, now I understand. Still, this code looks somewhat "hacky" to me, but I can't think about a way of improving it.